### PR TITLE
Bump imagemin version

### PR DESCRIPTION
**Pull request by kevva**
Was hoping to be able to cleanup the code some but couldn't find a nicer way to calculate the optimizations.

### CODE REVIEW

File: test.js [hunk 1]
```diff
@@ -22,7 +22,7 @@ const createFixture = async opts => {
 };
 
 test('minify images', async t => {
-	const {buf, stream} = await createFixture({optimizationLevel: 0});
```
Comment (kevva) [inline test.js]: Using `0` here caused OptiPNG to fail for some reason. All other levels works.

File: readme.md [hunk 2]
```diff
@@ -26,8 +25,7 @@ gulp.task('default', () => {
 			svgoPlugins: [
 				{removeViewBox: false},
 				{cleanupIDs: false}
-			],
-			use: [pngquant()]
```
Comment (kevva) [inline readme.md]: Removed this example since it'll confuse users and they'll complain when only their PNGs gets optimized.

File: package.json [hunk 3]
```diff
@@ -37,17 +37,23 @@
   "dependencies": {
     "chalk": "^1.0.0",
     "gulp-util": "^3.0.0",
-    "imagemin": "^4.0.0",
-    "object-assign": "^4.0.1",
+    "imagemin": "^5.0.0",
+    "imagemin-gifsicle": "^5.0.0",
+    "imagemin-mozjpeg": "^6.0.0",
+    "imagemin-optipng": "^5.0.0",
+    "imagemin-svgo": "^5.0.1",
```
Comment (sindresorhus) [inline package.json]: Maybe put these in optionalDependencies and handle them not being installed?

Comment (kevva) [inline package.json]: Yes.

**Comment by sindresorhus** [comment]
We discussed removing all the options and have some good defaults, if the user wants something different they can define the plugins manually. We would still expose the default ones on main export. Right?

File: .travis.yml [hunk 4]
```diff
@@ -1,6 +1,6 @@
 sudo: false
 language: node_js
 node_js:
+  - '6'
   - '5'
```
Comment (sindresorhus) [inline .travis.yml]: Remove

File: test.js (hunk 1 above)
Comment (sindresorhus) [inline test.js]: @kevva Maybe protect against that in the OptiPNG plugin then?

**Comment by kevva** [comment]
You mean like `module.exports.defaults = [...]` or `module.exports.mozjpeg = imageminMozjpeg`?

File: test.js (hunk 1 above)
Comment (kevva) [inline test.js]: Only happens with some images.

**Comment by sindresorhus** [comment]
The latter.

**Comment by sindresorhus** [comment]
If we do what I suggested we can just allow passing plugins as the first argument.

File: index.js [hunk 5]
```diff
@@ -32,60 +36,57 @@ module.exports = function (opts) {
...
-			}));
-
-		if (opts.use) {
-			opts.use.forEach(imagemin.use.bind(imagemin));
-		}
+		const use = plugins || [
+			imageminGifsicle(),
+			imageminMozjpeg(),
+			imageminOptipng({optimizationLevel: 3}),
+			imageminSvgo()
```
Comment (kevva) [inline index.js]: Are there any other defaults we want here? I don't want to force interlacing and such on people.

Comment (sindresorhus) [inline index.js]: Maybe `multipass`?

Comment (kevva) [inline index.js]: As long as it doesn't decrease performance too much. I'll benchmark it.

Comment (kevva) [inline index.js]:
```
Compare: multipass vs no multipass

        multipass
            √ Completed in 12.382s (107.969 ops/sec)
        no multipass
            √ Completed in 11.167s (189.531 ops/sec)

        √ Completed in 23.549s, no multipass was fastest by 43.034%
```

Comment (sindresorhus) [inline index.js]: Seems fine. Not much difference in speed.

Comment (kevva) [inline index.js]: What do you mean by "in speed"? I don't know how bencha works but the run time compared to the ops doesn't make sense. It's like it's running for a fixed amount of time or something.

Comment (sindresorhus) [inline index.js]: Yeah, that's weird. I still think it's worth it though. `optipng` also does multiple passes by default.

Comment (kevva) [inline index.js]: Ok, added it. Also fixed the docs.

File: readme.md [hunk 6]
```diff
@@ -38,74 +30,32 @@ gulp.task('default', () => {
...
-
-Type: `boolean`  
-Default: `false`
-
-Lossless conversion to progressive.
+#### plugins
 
-##### interlaced *(gif)*
+Type: `array`<br>
+Default: `[imageminGifsicle(), imageminMozjpeg(), imageminOptipng({optimizationLevel: 3}), imageminSvgo({multipass: true})]`
```
Comment (sindresorhus) [inline readme.md]: Prefix all the methods with `imagemin.`, so it will work in case the user chooses to copy-paste.

File: readme.md [hunk 7]
```diff
@@ -38,74 +30,32 @@ gulp.task('default', () => {
...
-
-Customize which SVGO plugins to use. [More here](https://github.com/sindresorhus/grunt-svgmin#available-optionsplugins).
-
-##### use
-
-Type: `array`  
-Default: `null`
-
-Additional [plugins](https://www.npmjs.com/browse/keyword/imageminplugin) to use with imagemin.
+Output more detailed information
```
Comment (sindresorhus) [inline readme.md]: `.`

**Comment by sindresorhus** [comment]
Maybe worth mentioning in the readme that it comes with good default and the user probably doesn't need to change it.

**Comment by kevva** [comment]
> Maybe worth mentioning in the readme that it comes with good default and the user probably doesn't need to change it.

Where do you want it?

**Comment by sindresorhus** [comment]
At the `API` section.

**Comment by kevva** [comment]
Alright, added a note about it. Please improve it if needs :).

**Comment by sindresorhus** [comment]
Looks good :)

**Pull request merged**